### PR TITLE
ceph: add mgr caps for csi-rbd-node client

### PR DESCRIPTION
**Pull request by Madhu-1**
**Description of your changes:**
If the kernel doesn't support mapping of the rbd image with the deep-flatten feature,cephcsi needs to flatten the rbd image first and then map the image on the node.

cephcsi first tries to add a task to flatten the rbd image if it receives any permission error it will try to call the rbd CLI command which is a blocking call. This commit adds mgr caps to the csi-rbd-node user so that cephcsi will add flatten task and return
immediate error to the kubelet, let kubelet retry again, If we go with blocking rbd CLI call we may end up having stale maps on the node in corner cases.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

**Review by leseb — COMMENTED** [review]
Please update this too https://github.com/rook/rook/blob/master/cluster/examples/kubernetes/ceph/create-external-cluster-resources.py#L514

**Review by leseb — CHANGES_REQUESTED** [review]
Please update this too https://github.com/rook/rook/blob/master/cluster/examples/kubernetes/ceph/create-external-cluster-resources.py#L514

File: cluster/examples/kubernetes/ceph/create-external-cluster-resources.py [hunk 1]
```diff
@@ -512,6 +512,7 @@ def create_cephCSIKeyring_RBDNode(self):
         cmd_json = {"prefix": "auth get-or-create",
                     "entity": "client.csi-rbd-node",
                     "caps": ["mon", "profile rbd",
+                             "mgr", "allow rw",
```
Comment (leseb) [inline cluster/examples/kubernetes/ceph/create-external-cluster-resources.py:515]: @aruniiird can you suggest what to change as well when the `--upgrade` flag is passed? Thanks

Comment (Madhu-1) [reply cluster/examples/kubernetes/ceph/create-external-cluster-resources.py:515]: @leseb this script has the support to update the caps internally if the user already exists with different cap?

Comment (leseb) [reply cluster/examples/kubernetes/ceph/create-external-cluster-resources.py:515]: Yes, it does, that's why I'm asking @aruniiird who wrote the update logic recently to advise.

Comment (aruniiird) [reply cluster/examples/kubernetes/ceph/create-external-cluster-resources.py:515]: @leseb , @Madhu-1 , in order to run upgrade you require an existing user and if you have multiple pool-prefixes you need to specify that as well. So the command may look some what like below,

python3 create-external-cluster-resources.py --upgrade --run-as-user | -u <user-name>  [ --rgw-pool-prefix <pool-prefix> ]

Most of the time you don't have to provide the `<pool-prefix>` as the script will try to internally figure out the pool prefix name.

File: pkg/operator/ceph/csi/secrets.go [hunk 2]
```diff
@@ -81,6 +81,7 @@ func createCSIKeyringCephFSProvisioner(s *keyring.SecretStore) (string, error) {
 func cephCSIKeyringRBDNodeCaps() []string {
 	return []string{
 		"mon", "profile rbd",
+		"mgr", "allow rw",
```
Comment (travisn) [inline pkg/operator/ceph/csi/secrets.go:84]: If an upgraded cluster, will the new caps be applied?

Comment (Madhu-1) [reply pkg/operator/ceph/csi/secrets.go:84]: Yes @travisn that is handled.

File: cluster/examples/kubernetes/ceph/create-external-cluster-resources.py (hunk 1 above)
Comment (Madhu-1) [reply cluster/examples/kubernetes/ceph/create-external-cluster-resources.py:515]: @aruniiird to be more precise do `create-external-cluster-resources.py` is capable of updating the user caps if the user already exists?

**Review by travisn — APPROVED** [review]

**Comment by travisn** [comment]
Looks like the canary integration test is failing with this error in the script check:
```
  File "/etc/ceph/create-external-cluster-resources.py", line 568, in _gen_output_map

    self.out_map['CSI_RBD_NODE_SECRET_SECRET'] = self.create_cephCSIKeyring_RBDNode()
Exception thrown successfully: Out of range port number: 90320
  File "/etc/ceph/create-external-cluster-resources.py", line 518, in create_cephCSIKeyring_RBDNode
Exception thrown successfully: Out of range port number: 73422
    ret_val, json_out, err_msg = self._common_cmd_json_gen(cmd_json)
Exception thrown successfully: 'prometheus' service port must listen on 9283. You can change it with 'ceph config set mgr mgr/prometheus/server_port 9283'.
  File "/etc/ceph/create-external-cluster-resources.py", line 278, in _common_cmd_json_gen
```

File: cluster/examples/kubernetes/ceph/create-external-cluster-resources.py (hunk 1 above)
Comment (Madhu-1) [reply cluster/examples/kubernetes/ceph/create-external-cluster-resources.py]: currently, the script is not having any option to update the existing user caps created https://github.com/rook/rook/issues/6792 to track python changes.

**Comment by Madhu-1** [comment]
@travisn removed python script changes from the PR, it will be handled in separate PR as script need to be enhanced to update the existing users, @aruniiird will be working on the Part.

**Review by travisn — APPROVED** [review]

**Comment by Madhu-1** [comment]
@travisn PTAL

**Pull request merged**